### ISSUE TITLE
How to capture a screen shot from webRTC video feed

### ISSUE THREAD

**Issue by Mathew-Harvey** [AirenSoft/OvenPlayer#234]
Hi, 

I'm trying to capture a screenshot (write it to a canvas) from a streaming webRTC feed. I'm rendering the stream in the browser like this:

 const player = OvenPlayer.create(id="player_id", {
            sources: [
                {
                    label: 'marinestream',
                    // Set the type to 'webrtc'
                    type: 'webrtc',
                    // Set the file to WebRTC Signaling URL with OvenMediaEngine 
                    file: 'ws://192.168.50.248:3333/app/stream'
                }
            ]
        });

Previously I tried to grab the div that the player is located in (id="player_id"). However, this didn't work, the system doesn't see HTMLVideoElement within the div.

How can I access the HTMLVideoElement of the player?

**Comment by Mathew-Harvey** [comment]
ok, figured it out:

<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="UTF-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>webRTC streaming and capture test</title>
</head>

<body>
    
    <div style="height: 720px; width: 1280px;">
        <div style="height: 720px;" id="player_id"></div>
    </div>
    
    <button id="screenshot-button"> Screen Shot </button>

    
    <img id="screenshot-img" src="">
    <canvas style="display:none;"></canvas>

    
    <script src="https://cdn.jsdelivr.net/npm/ovenplayer/dist/ovenplayer.js"></script>
    <script src="https://webrtc.github.io/adapter/adapter-latest.js"></script>
    <script>

        // Initialize OvenPlayer
        const player = OvenPlayer.create(id="player_id", {
            sources: [
                {
                    label: 'marinestream',
                    // Set the type to 'webrtc'
                    type: 'webrtc',
                    // Set the file to WebRTC Signaling URL with OvenMediaEngine 
                    file: 'ws://192.168.50.248:3333/app/stream'
                }
            ]
        });

        const screenshotButton = document.querySelector("#screenshot-button");
        const img = document.querySelector("#screenshot-img");
        const videoWrapper = document.getElementsByClassName("op-media-element-container");
        
        const canvas = document.createElement("canvas");

        const getVideo = (className) => {
            const videoWrapper = document.getElementsByClassName(className);
            const wrapperFound = videoWrapper !== undefined && videoWrapper.length > 0;
            if (wrapperFound === false) {
                console.log("Wrapper not found", {videoWrapper});
                return undefined;
            }
            const videoElementFound = videoWrapper[0].children?.length > 0;
            if(!videoElementFound) {
                console.log("Video Element not found", {videoWrapper: videoWrapper});
                return undefined;
            }
            return videoWrapper[0].children[0];
        };      
        screenshotButton.onclick = function () {
            
            const video = getVideo("op-media-element-container")
            
            console.log("video dom element", { video , videoWrapper, width: video.videoWidth, height: video.videoHeight })
            
            canvas.width = video.videoWidth;
            canvas.height = video.videoHeight;
            canvas.getContext("2d").drawImage(video, 0, 0, video.videoWidth, video.videoHeight);

            img.src = canvas.toDataURL("image/webp");
        };

     

        

    </script>

</body>

**Comment by SangwonOh** [comment]
@Mathew-Harvey Hi! Great work!

Since OvenPlayer contains only one video element in the player wrapper (in your case "player_id"),  You can simply get the video element like below.

```
const video = document.getElementById('player_id').getElementsByTagName('video')
```

**Comment by Mathew-Harvey** [comment]
@SangwonOh - Hi!

We tried that : const video = document.getElementById('player_id').getElementsByTagName('video')

however it returned this error:

Uncaught TypeError: Failed to execute 'drawImage' on 'CanvasRenderingContext2D': The provided value is not of type '(CSSImageValue or HTMLCanvasElement or HTMLImageElement or HTMLVideoElement or ImageBitmap or OffscreenCanvas or SVGImageElement or VideoFrame)'.
    at HTMLButtonElement.screenshotButton.onclick

So basically when I clicked the button, it didn't recognise the const video as a video element... hence we had to do some DOM crawling to figure out how to grab the video element.
